src/TwigComponent/src/ComponentTemplateFinder.php

```diff
@@ -42,28 +42,27 @@ public function findAnonymousComponentTemplate(string $name): ?string
 
         // Legacy auto-naming rules < 2.13
         if (null === $this->directory) {
-            if ($loader->exists('components/'.$componentPath.'.html.twig')) {
-                return 'components/'.$componentPath.'.html.twig';
-            }
-
-            if ($loader->exists($componentPath.'.html.twig')) {
-                return $componentPath.'.html.twig';
-            }
-
-            if ($loader->exists('components/'.$componentPath)) {
-                return 'components/'.$componentPath;
-            }
-
-            if ($loader->exists($componentPath)) {
-                return $componentPath;
+            foreach ([
+                'components/'.$componentPath.'.html.twig',
+                'components/'.$componentPath.'.twig',
+                $componentPath.'.html.twig',
+                $componentPath.'.twig',
+                'components/'.$componentPath,
+                $componentPath,
+            ] as $path) {
+                if ($loader->exists($path)) {
+                    return $path;
+                }
             }
 
             return null;
         }
 
-        $template = rtrim($this->directory, '/').'/'.$componentPath.'.html.twig';
-        if ($loader->exists($template)) {
-            return $template;
+        foreach (['.html.twig', '.twig'] as $extension) {
```

Comment:
I dont think we want to accept anonymous Component with multiple extensions. The idea was to have a 1:1 relation betwee  an anonymous template and its name..